python/interpret-core/interpret/glassbox/_ebm/_ebm.py

```diff
@@ -2145,6 +2145,129 @@ def monotonize(self, term, increasing="auto", passthrough=0.0):
 
         return self
 
+    def remove_terms(self, term_list):
+        """Removes terms (and their associated components) from a fitted EBM. Note 
+        that this will change the structure (i.e., by removing the specified 
+        indices) of the following components of ``self``: ``term_features_``,
+        ``term_names_``, ``term_scores_``, ``bagged_scores_``, 
+        ``standard_deviations_``, and ``bin_weights_``.
+
+        Args:
+            term_list: A list of term names or indices.
+
+        Returns:
+            Itself.
+        """
+        check_is_fitted(self, "has_fitted_")
+
+        # If term_list contains term names, convert them to indices
+        term_list = [self.term_names_.index(term) if isinstance(term, str) 
+                    else term for term in term_list]
+
+        # Copy any fields we'll overwrite in case someone has a shallow copy of self
```

Comment:
These copies are not required in this case since we later use a list comprehension which will create a new list.